cmake/modules/FindGnuLd.cmake

```diff
@@ -23,13 +23,30 @@
 # Note that this will use CROSS_COMPILE, if defined,
 # as a prefix to the linker executable.
 
+# GNULD_LINKER exists on repeated builds...
+if(EXISTS "${GNULD_LINKER}")
+  if(DEFINED GNULD_LINKER_IS_BFD)
+    # ... no need to redo the below again.
+    return()
+  else()
```

Comment:
The `return()` seems unneeded, why not just `if(NOT DEFINED GNULD_LINKER_IS_BFD)`?

Reply:
It is because since `GNULD_LINKER` is defined, the next big if-block would go into the first true block (line 49) and set `GNULD_LINKER_IS_BFD` to on/true (regardless of whether linker is bfd compatible).

Reply:
I'm not sure I understand... are you just saying you want to keep the order the same as line 48+ ?

Reply:
If we don't return early, the next time this module runs, it would go through the first true block after `execute_process()` as `GNULD_LINKER` is already defined and `GNULD_LINKER_IS_BFD` is being set to true regardless whether linker is bfd compatible. It happened when I tried to use xcc with or without `FORCE` in `set()`. Removing `GNULD_LINKER` doesn't seem to help in this situation. Hence the need to return early.

Reply:
Sorry I totally missed that `return()` aborts the entire file. My bad, sorry for the noise.